sklearn/metrics/pairwise.py

```diff
@@ -920,6 +953,9 @@ def paired_cosine_distances(X, Y):
     'manhattan': paired_manhattan_distances,
     'cityblock': paired_manhattan_distances}
 
+PAIRED_DISTANCES_WITHOUT_RESTRICTIONS = {
```

Comment:
I don't understand why this was introduced in the public API. I understand if haversine needs distinguishing from others in testing, but why here? I also don't understand what "without restrictions" intends, when haversine is restricted to dense and 2 features